src/main/java/io/vertx/redis/impl/AbstractRedisService.java

```diff
@@ -0,0 +1,245 @@
+            replyMessage.put("value", message);
+            eb.send(vertxChannel, replyMessage);
+          });
+        }
+        break;
+      // argument ["pattern" ["pattern"...]]
+      case "punsubscribe":
+        // unregister all channels
+        if (redisArgs == null || redisArgs.size() == 0) {
+          // unsubscribe all
+          expectedReplies = subscriptions.patternSize();
+          subscriptions.unregisterPatternSubscribeHandler(null);
+        } else {
+          expectedReplies = redisArgs.size();
+          for (Object obj : redisArgs) {
+            String pattern = (String) obj;
+            subscriptions.unregisterPatternSubscribeHandler(pattern);
+          }
+        }
+        break;
+      // argument ["channel" ["channel"...]]
+      case "unsubscribe":
+        // unregister all channels
+        if (redisArgs == null || redisArgs.size() == 0) {
+          // unsubscribe all
+          expectedReplies = subscriptions.channelSize();
+          subscriptions.unregisterChannelSubscribeHandler(null);
+        } else {
+          expectedReplies = redisArgs.size();
+          for (Object obj : redisArgs) {
+            String channel = (String) obj;
+            subscriptions.unregisterChannelSubscribeHandler(channel);
+          }
+        }
+        break;
+    }
+
+    redisClient.send(new Command(command, redisArgs, charset).setExpectedReplies(expectedReplies).setHandler(reply -> {
+      switch (reply.type()) {
+        case '-': // Error
+          resultHandler.handle(new RedisAsyncResult<>(reply.asType(String.class)));
+          return;
+        case '+':   // Status
+          resultHandler.handle(new RedisAsyncResult<>(null, reply.asType(returnType)));
+          return;
+        case '$':  // Bulk
+          if (transform == ResponseTransform.INFO) {
+            String info = reply.asType(String.class, encoding);
+            String lines[] = info.split("\\r?\\n");
+            JsonObject value = new JsonObject();
+            JsonObject section = null;
+            for (String line : lines) {
+              if (line.length() == 0) {
+                // end of section
+                section = null;
+                continue;
+              }
+
+              if (line.charAt(0) == '#') {
+                // begin section
+                section = new JsonObject();
+                // create a sub key with the section name
+                value.put(line.substring(2).toLowerCase(), section);
+              } else {
+                // entry in section
+                int split = line.indexOf(':');
+                if (section == null) {
+                  value.put(line.substring(0, split), line.substring(split + 1));
+                } else {
+                  section.put(line.substring(0, split), line.substring(split + 1));
+                }
+              }
+            }
+            resultHandler.handle(new RedisAsyncResult<>(null, (T) value));
```

Comment:
I believe we can get rid of RedisAsyncResult and use Future.succeededFuture or Future.failedFuture